debug_toolbar/middleware.py

```diff
@@ -48,7 +49,12 @@ def __call__(self, request):
         # Decide whether the toolbar is active for this request. Don't render
         # the toolbar during AJAX requests.
         show_toolbar = get_show_toolbar()
-        if not show_toolbar(request) or request.is_ajax():
+        accepts_html = (
+            not request.is_ajax()
+            if django.VERSION < (3, 1)
+            else request.accepts("text/html")
+        )
```

Comment:
Can you rejig this to only check `accepts_html` in the case of `show_toolbar` returning `True`? Perhaps split it to another function call. This would keep the `DEBUG=False` pathway minimal.

Reply:
Good call, thanks.

Now short circuits on `not show_toolbar(request)`.